File: Volume.py
import math

class Cube:
    def __init__(self, ndim=2, len=1.0):
        self.ndim = ndim
        self.len = len

    def CVolume(self):
        return math.pow(self.len, self.ndim)

    def print_info(self):
        print(f"Dimension: {self.ndim}, Length: {self.len}, Volume of the Cube: {self.CVolume()}")

class Sphere:
    def __init__(self, ndim=2, rad=1.0):
        self.ndim = ndim
        self.rad = rad

    def SVolume(self):
        return (math.pow(math.pi, self.ndim/2)*math.pow(self.rad, self.ndim) ) / math.gamma(self.ndim/ 2 + 1)

    def print_info(self):
        print(f"Dimension: {self.ndim}, Radius: {self.rad}, Volume of the Sphere: {self.SVolume()}")

File: Volume.cpp
#include "Volume.hpp"
#include <cstdio>
#include <cmath>

Cube::Cube(int ndim, double len): ndim(ndim), len(len){}

int Cube::Ndim() {
    return ndim;
}

double Cube::Len() {
    return len;
}

void Cube::Ndim(int ndim) {
    this->ndim = ndim;
}

void Cube::Len(double len) {
    this->len = len;
}

double Cube::CVolume() {
    return pow(len, ndim);
}
